### DISCUSSION TITLE
[BE] LOG 레벨 기준, 작성할 코드의 위치, 로그 포맷

### DISCUSSION THREAD

**Discussion by yenawee** [woowacourse-teams/2023-stamp-crush#398] [General]
info, warn, error 등의 로그를 어느 위치에 어떤 걸 찍을지 결정합니다

**Comment by gitchannn** [comment]
# 결정사항

## 로그 레벨의 기준

- **ERROR**: 500대
- **WARN**: 400대
- **INFO**: 중요한 비즈니스 로직이 성공했을 때 _(예. GITCHAN 고객이 JENA CAFE의 Stamp 적립에 성공했습니다!)_

## 로그 작성할 코드 내 위치

- 레이어 상관 없이 필요한 곳에서 `log.warn`, `log.info`로 작성한다. 
- `log.error`는 우리가 잡지 못한 Exception만을 잡을 예정이기 때문에, ExceptionHandler에서만 처리한다.

**Comment by gitchannn** [comment]
WARNING 로그를 어디에 찍을지

**Reply by youngh0** [reply]
service에서 바로 찍어도 좋을 것 같아요. 로그는 모든 layer의 관심사인 횡단 관심사인 것 같아서 프로덕션 코드와 섞이는게 어색하진 않은 것 같습니다.

오히려 로그를 위해서 모든 커스텀 exception을 고치는 것도 어려워 보이고, 모든 예외를 다 커스텀 해야할 것 같아요. 스프링에서 제공하는 예외에는 id값을 넣을 수 없으니까요.

**Reply by gitchannn** [reply]
Service, Respository에서 바로 찍자! 
최대한 꼼꼼히 어떤 내용을 담을지는 자세히 정하면 좋을 듯

**Reply by jundonghyuk** [reply]
코드내에 인라인으로 로그를 찍는게 좋다는 의견입니다.

로깅 포맷이 다 다를 수 있는데 하나의 스태틱한 문자열로 관리하나요 ?

아니면 커스텀예외 안에도 넣을 수 있을거같거여

**Reply by yenawee** [reply]
저는 ExceptionHandler의 사용 의미가 에러 잡아서 log 도 찍어주는데 있다고 봤어서 ExceptionHandler 에서 처리해주는게 좋다고 봅니다
그리고 log 가 코드 여기저기 찍혀있는게 깔끔하지가 않을거같아서..

근데 log.info() 가 어차피 여기저기 찍힐테니까 딱 ExceptionHandler 에서 꼭 해야한다는 의견은 아닙니다

**Comment by gitchannn** [comment]
로그의 포맷 
- 로그를 보는 때가 오면, 심각한 상황일 수 있으니 통일성 있는 포맷이 필요하다!

**Reply by gitchannn** [reply]
- 인증, 인가 못한 경우
- Cafe Not Found같이 자원을 못 찾은 경우
- 비즈니스 로직 상에서의 오류
